connector/controllers/main.py

```diff
@@ -60,28 +86,17 @@ def retry_postpone(job, message, seconds=None):
                 return ""
 
         try:
-            # if the job has been manually set to DONE or PENDING,
-            # or if something tries to run a job that is not enqueued
-            # before its execution, stop
-            if job.state != ENQUEUED:
-                _logger.warning('job %s is in state %s '
-                                'instead of enqueued in /runjob',
-                                job_uuid, job.state)
-                return
-
-            with session_hdl.session() as session:
-                # TODO: set_started should be done atomically with
-                #       update queue_job set=state=started
-                #       where state=enqueid and id=
-                job.set_started()
-                self.job_storage_class(session).store(job)
-
-            _logger.debug('%s started', job)
-            with session_hdl.session() as session:
-                job.perform(session)
-                job.set_done()
-                self.job_storage_class(session).store(job)
-            _logger.debug('%s done', job)
+            try:
+                self._try_perform_job(session_hdl, job)
+            except OperationalError as err:
+                # Automatically retry the typical transaction serialization
+                # errors
+                if err.pgcode not in PG_CONCURRENCY_ERRORS_TO_RETRY:
+                    raise
+
+                session.env.clear()
```

Comment:
Note: the session variable here is surely closed as it is used outside its context manager. Not sure it is a problem though. Maybe we should put the OperationalError except inside of a session_hdl.session context manager and pass the session to try_perform_job instead of the handler, thus we would act on the correct env. I can try that Monday.